beta-src/src/components/ui/WDMessageList.tsx

```diff
@@ -24,27 +23,53 @@ const WDMessageList: React.FC<WDMessageListProps> = function ({
   countryIDSelected,
   messagesEndRef,
 }): React.ReactElement {
-  const [viewport] = useViewport();
-  const device = getDevice(viewport);
-  const height = "350px";
+  const viewedPhaseState = useAppSelector(gameViewedPhase);
+
   const filteredMessages = messages.filter(
     (message) =>
       (message.fromCountryID === countryIDSelected ||
         message.toCountryID === countryIDSelected) &&
       (countryIDSelected === 0 || message.toCountryID !== 0), // public messages in public chat
   );
-  const messageComponents = filteredMessages.map((message: GameMessage) => (
-    <WDMessage
-      key={`${message.timeSent}:${message.fromCountryID}:${message.toCountryID}:${message.message}`}
-      message={message}
-      userCountry={userCountry}
-      allCountries={allCountries}
-    />
-  ));
+  const messagesByTurn: { [key: number]: React.ReactElement[] } = {};
+  filteredMessages.forEach((message: GameMessage) => {
```

Comment:
I think this would be easier to test and reason about as a `map`, as you're immediately about to iterate over the keys of the temporary object that is being constructed here.